pkg/controllers/cisnetworkpolicy/controller.go

```diff
@@ -0,0 +1,166 @@
+	k8s kubernetes.Interface
+}
+
+func (h *handler) onRemove(key string, node *core.Node) (*core.Node, error) {
+	return h.reconcileFlannelHostNetworkPolicy(key, node)
+}
+
+func (h *handler) onChange(key string, node *core.Node) (*core.Node, error) {
+	if node == nil {
+		return nil, nil
+	}
+
+	return h.reconcileFlannelHostNetworkPolicy(key, node)
+}
+
+func (h *handler) reconcileFlannelHostNetworkPolicy(nodeName string, node *core.Node) (*core.Node, error) {
+	var np *netv1.NetworkPolicy
+	var npIR *netv1.NetworkPolicyIngressRule
+
+	npIR, err := h.generateHostNetworkPolicyIngressRule()
+	if err != nil {
+		return nil, err
+	}
+
+	np, err = h.generateHostNetworkingNetworkPolicy(npIR)
+	if err != nil {
+		return nil, err
+	}
+
+	var namespaces = []string{
+		metav1.NamespaceSystem,
+		metav1.NamespaceDefault,
+		metav1.NamespacePublic,
+	}
+
+	for _, namespace := range namespaces {
+		applied := false
+		for !applied {
+			if npGet, err := h.k8s.NetworkingV1().NetworkPolicies(namespace).Get(h.ctx, FlannelHostNetworkPolicyName, metav1.GetOptions{}); err != nil {
+				if errors.IsNotFound(err) {
+					if _, err := h.k8s.NetworkingV1().NetworkPolicies(namespace).Create(h.ctx, np, metav1.CreateOptions{}); err != nil {
+						if !errors.IsAlreadyExists(err) {
+							logrus.Errorf("CISNetworkPolicyController: Error creating network policy in namespace %s: %v", namespace, err)
+							return nil, err
+						}
+					}
+				} else {
+					logrus.Errorf("CISNetworkPolicyController: Error getting network policy network policy in namespace %s: %v", namespace, err)
+					return nil, err
+				}
+			} else {
+				npGet = npGet.DeepCopy()
+				npGet.Spec.Ingress[0] = *npIR
+				_, err := h.k8s.NetworkingV1().NetworkPolicies(namespace).Update(h.ctx, npGet, metav1.UpdateOptions{})
+				if err != nil {
+					if errors.IsConflict(err) {
+						logrus.Errorf("CISNetworkPolicyController: Conflict updating network policy network policy in namespace %s: %v", namespace, err)
+						continue
+					}
+				}
+				applied = true
+
+			}
+		}
+	}
+	logrus.Debugf("CISNetworkPolicyController: Handled node change")
+	return nil, nil
+}
+
+func (h *handler) generateHostNetworkingNetworkPolicy(networkPolicyIngressRule *netv1.NetworkPolicyIngressRule) (*netv1.NetworkPolicy, error) {
+	np := &netv1.NetworkPolicy{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: FlannelHostNetworkPolicyName,
+		},
+		Spec: netv1.NetworkPolicySpec{
+			PodSelector: metav1.LabelSelector{},
+			Ingress: []netv1.NetworkPolicyIngressRule{
+				*networkPolicyIngressRule,
+			},
+			PolicyTypes: []netv1.PolicyType{
+				netv1.PolicyTypeIngress,
+			},
+		},
+	}
+
+	return np, nil
+}
+
+func (h *handler) generateHostNetworkPolicyIngressRule() (*netv1.NetworkPolicyIngressRule, error) {
+	var npIR netv1.NetworkPolicyIngressRule
+
+	nodes, err := h.k8s.CoreV1().Nodes().List(h.ctx, metav1.ListOptions{})
+	if err != nil {
+		logrus.Infof("CISNetworkPolicyController: problem listing nodes %v", err)
```

Comment:
Should this be error or debug, or just not logged at all? I second Hussein's comment about not logging errors if we're also returning them.

Reply:
This should be an error, but if you look at our internal thread on this, I have a reason for specifically calling `Errorf`.

Reply:
`logrus.Warnf` ?

Reply:
It's an error, and should be stated as such.

Reply:
`. o ( bubble it up? )`

Reply:
It's an error now and bubbled up